### PR TITLE
Improve ES query speed by using traceId

### PR DESCRIPTION
**Pull request by zifeihan**
### Improve ES query speed by using traceId
- [ ] Add a benchmark for the improvement, refer to [the existing ones](https://github.com/apache/skywalking/blob/master/apm-commons/apm-datacarrier/src/test/java/org/apache/skywalking/apm/commons/datacarrier/LinkedArrayBenchmark.java)
- [ ] The benchmark result.
```text
<Paste the benchmark results here>
```
- [ ] Links/URLs to the theory proof or discussion articles/blogs. <links/URLs here>

Because the third part of traceId has time information, we can search within the specified time slice when querying. When there are many fragments and the amount of data is large, this can greatly improve the query speed.

### CODE REVIEW

File: oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/TraceQueryEsDAO.java [hunk 1]
```diff
@@ -162,7 +163,11 @@ public TraceBrief queryBasicTraces(long startSecondTB,
         sourceBuilder.query(QueryBuilders.termQuery(SegmentRecord.TRACE_ID, traceId));
         sourceBuilder.size(segmentQueryMaxSize);
 
-        SearchResponse response = getClient().search(SegmentRecord.INDEX_NAME, sourceBuilder);
+        String traceIdThirdPart = traceId.split("\\.")[2];
+        long timeBucket = TimeBucket.getRecordTimeBucket(Long.valueOf(traceIdThirdPart) / 10000);
+
+        SearchResponse response = getClient().search(
+            new TimeRangeIndexNameMaker(SegmentRecord.INDEX_NAME, timeBucket, timeBucket), sourceBuilder);
```
Comment (wu-sheng) [inline oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/TraceQueryEsDAO.java:170]: I am afraid this is not working. We don't have this timestamp-based trace-id rule anymore.

Comment (wu-sheng) [reply oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/TraceQueryEsDAO.java:170]: Take a look at other agent implementations, such as LUA.

Comment (zifeihan) [reply oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/TraceQueryEsDAO.java:170]: > Take a look at other agent implementations, such as LUA.

Ok let me take a look.

Comment (zifeihan) [reply oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/TraceQueryEsDAO.java]: @wu-sheng I modified some content, can you help me see if this is feasible?

Comment (wu-sheng) [reply oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/TraceQueryEsDAO.java]: From my personal understanding, we don't have the format rule of traceid anymore. Any string is acceptable, and even it is combined with 3 parts, doesn't mean there is a timestamp included.

**Pull request closed without merging**